include/seqan3/alphabet/composition/union_composition.hpp

```diff
@@ -461,39 +330,202 @@ class union_composition
 protected:
     //!\privatesection
 
-    //!\brief Converts an object of one of the given alphabets into the internal representation
+    /*!\brief Returns the max over all alphabet_size_v<alphabets_t>'s.
+     * \tparam ...alphabets_t The types must satisfy seqan3::alphabet_concept.
+     *
+     * ```cpp
+     * constexpr size_t max = max_of_alphabet_sizes_v<dna4, gap, dna5>;
+     * assert(max == 5);
+     * ```
+     */
+    template <typename ...alphabets_t>
+    //!\cond
+        requires (alphabet_concept<alphabets_t> && ...)
+    //!\endcond
+    static constexpr size_t max_of_alphabet_sizes_v =
+        std::max({static_cast<size_t>(0), static_cast<size_t>(alphabet_size_v<alphabets_t>)...});
+
+    /*!\brief Returns an array which contains the prefix sum over all alphabet_types::value_size's.
+     * \tparam ...alphabet_types The types must satisfy seqan3::alphabet_concept.
+     *
+     * ```cpp
+     * constexpr std::array partial_sum = partial_sum_of_alphabet_sizes<dna4, gap, dna5>();
+     * assert(partial_sum.size() == 4);
+     * assert(partial_sum[0] == 0);
+     * assert(partial_sum[1] == 4);
+     * assert(partial_sum[2] == 5);
+     * assert(partial_sum[3] == 10);
+     * ```
+     */
+    template <typename ...alphabets_t>
+    //!\cond
+        requires (alphabet_concept<alphabets_t> && ...)
+    //!\endcond
+    static constexpr auto partial_sum_of_alphabet_sizes() noexcept
+    {
+        constexpr size_t value_size = sum_of_alphabet_sizes_v<alphabets_t...>;
+        using rank_t = detail::min_viable_uint_t<value_size>;
+
+        constexpr size_t N = sizeof...(alphabets_t) + 1;
+        using array_t = std::array<rank_t, N>;
+
+        array_t partial_sum{0, alphabet_size_v<alphabets_t>...};
+        for (size_t i = 1u; i < N; ++i)
+            partial_sum[i] += partial_sum[i-1];
+
+        return partial_sum;
+    }
+
+    /*!\brief Returns an fixed sized map at compile time where the key is the rank
+     * of alphabet_t and the value is the corresponding char of that rank.
+     * \sa value_to_char_table
+     *
+     * ```cpp
+     * constexpr std::array table1 = make_value_to_char_table<5, char>(dna4{});
+     * assert(table1.size() == 5);
+     * assert(table1[0] == 'A');
+     * assert(table1[1] == 'C');
+     * assert(table1[2] == 'G');
+     * assert(table1[3] == 'T');
+     * assert(table1[4] == '\0');
+     * ```
+     */
+    template <size_t table_size, typename alphabet_t>
+        requires (alphabet_size_v<alphabet_t> <= table_size)
+    static constexpr auto make_value_to_char_table(alphabet_t alphabet) noexcept
```

Comment:
This is called `make_value_to_char_table` so does it make the `value_to_char_table` ?

Reply:
It makes a `value_to_char_table` for a specific alphabet, but with a fixed-sized table. It is similar to https://github.com/seqan/seqan3/blob/b9eb19c183160f58da806cb619fd66c1519a5ee5/include/seqan3/alphabet/detail/convert.hpp#L66.

The name was `value_to_char_table_I` before, because it will be called in the i-th "iterator" of `value_to_char_table()`. Jörg thought this name was non conforming so I changed it to `make_value_to_char_table`, because it creates a fixed-size value_to_char_table for one alphabet.